components/reusable-components/reusable-form/ReusableForm.tsx

```diff
@@ -0,0 +1,193 @@
+    firstName: string;
+    lastName: string;
+    email: string;
+    password: string;
+  }> = useFormik({
+    initialValues,
+    validationSchema,
+    onSubmit: (values) => {
+      setPasswordValidation({
+        uppercase: /[A-Z]/.test(values.password) ? true : false,
+        specialChar: /[!@#$%^&*(),.?":{}|<>]/.test(values.password) ? true : false,
+        minLength: values.password.length >= 8 ? true : false,
+      });
+    },
+  });
+
+  const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setPassword(value);
+    formik.handleChange(e);
+    const newChange = {
+      uppercase: /[A-Z]/.test(value) ? true : null,
+      specialChar: /[!@#$%^&*(),.?":{}|<>]/.test(value) ? true : null,
+      minLength: value.length >= 8 ? true : null,
+    };
+    setPasswordValidation((prevState) => ({
+      ...prevState,
+      ...newChange,
+    }));
+  };
+
+  const getColor = (valid: boolean | null) => {
+    return valid === null ? '#475569' : valid ? '#00B107' : '#FF4045';
+  };
+
+  return (
+    <div
+      className={`${style.reusableFormWrapper} ${lightTheme ? style.lightFormWrapper : style.darkFormWrapper}`}
+    >
+      <h2 className={style.formTitle}>Регистрирај се</h2>
+      <form className={style.reusableForm} onSubmit={formik.handleSubmit}>
+        <TextInput
+          placeholder="Внесете го вашето име"
+          type="text"
+          label="Име *"
+          name="firstName"
+          field="firstName"
+          formik={formik}
+        />
+        <TextInput
+          placeholder="Внесете го вашето презиме"
+          type="text"
+          label="Презиме *"
+          name="lastName"
+          field="lastName"
+          formik={formik}
+        />
+        <TextInput
+          placeholder="Внесете ја вашата електронска пошта"
+          type="email"
+          label="Електронска пошта *"
+          name="email"
+          field="email"
+          formik={formik}
+        />
+        <TextInput
+          placeholder="Внесете ја вашата лозинка"
+          type="password"
+          label="Лозинка *"
+          name="password"
+          field="password"
+          formik={formik}
+          onChange={handlePasswordChange}
+        />
+        <div className={style.requirements}>
+          {Object.entries(passwordValidation).map(([rule, valid]) => (
+            <div className={style.requirementsItem} key={rule}>
+              <CheckPasswordValidityIcon color={getColor(valid)} />
+              <p>
+                {rule === 'uppercase'
+                  ? 'Една голема буква'
+                  : rule === 'specialChar'
+                    ? 'Еден специјален знак или симбол'
+                    : 'Минимум 8 карактери'}
+              </p>
+            </div>
+          ))}
+        </div>
+
+        <label htmlFor="checkbox">
+          <input type="checkbox" id="checkbox" />
+          Се согласувам со сите Terms & Conditions
+        </label>
```

Comment:
This needs to be required. The user shouldn't be able to Sign Up without checking this.